collect_app/src/main/java/org/odk/collect/android/preferences/AdminPreferencesFragment.java

```diff
@@ -53,20 +56,25 @@ public boolean onPreferenceClick(Preference preference) {
 
             LayoutInflater factory = LayoutInflater.from(getActivity());
             final View dialogView = factory.inflate(R.layout.password_dialog_layout, null);
-
             final EditText passwordEditText = (EditText) dialogView.findViewById(R.id.pwd_field);
-            final EditText verifyEditText = (EditText) dialogView.findViewById(R.id.verify_field);
-
+            final CheckBox passwordCheckBox = (CheckBox) dialogView.findViewById(R.id.checkBox2);
+            passwordCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                @Override
+                public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
+                    if (!passwordCheckBox.isChecked()) {
+                        passwordEditText.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_PASSWORD);
+                    } else {
+                        passwordEditText.setInputType(InputType.TYPE_TEXT_VARIATION_PASSWORD);
+                    }
+                }
+            });
             builder.setTitle(R.string.change_admin_password);
             builder.setView(dialogView);
             builder.setPositiveButton(getString(R.string.ok), new DialogInterface.OnClickListener() {
                 @Override
                 public void onClick(DialogInterface dialog, int which) {
                     String pw = passwordEditText.getText().toString();
-                    String ver = verifyEditText.getText().toString();
-
-                    if (!pw.equalsIgnoreCase("") && !ver.equalsIgnoreCase("") && pw.equals(ver)) {
-                        // passwords are the same
+                    if (!pw.equalsIgnoreCase("")) {
```

Comment:
Since we are checking if `pw` is not null, then `equalsTgnoreCase` can be replaced with `equals`